### ISSUE TITLE
Evaluation alphabets acuracy: 1.0, is that right?

### ISSUE THREAD

**Issue by basaltzhang** [Goldesel23/Siamese-Networks-for-One-Shot-Learning#1]
I changed the max train iteration to 18000, and after the training process the evaluation result shows below:

> Making One Shot Task on evaluation alphabets:
Atemayar_Qelisayer alphabet, accuracy: 1.0
ULOG alphabet, accuracy: 1.0
Sylheti alphabet, accuracy: 1.0
Angelic alphabet, accuracy: 1.0
Glagolitic alphabet, accuracy: 1.0
Ge_ez alphabet, accuracy: 1.0
Tengwar alphabet, accuracy: 1.0
Oriya alphabet, accuracy: 1.0
Avesta alphabet, accuracy: 1.0
Kannada alphabet, accuracy: 1.0
Aurek-Besh alphabet, accuracy: 1.0
Keble alphabet, accuracy: 1.0
Mongolian alphabet, accuracy: 1.0
Gurmukhi alphabet, accuracy: 1.0
Manipuri alphabet, accuracy: 1.0
Malayalam alphabet, accuracy: 1.0
Atlantean alphabet, accuracy: 1.0
Old_Church_Slavonic_(Cyrillic) alphabet, accuracy: 1.0
Tibetan alphabet, accuracy: 1.0
Syriac_(Serto) alphabet, accuracy: 1.0

> Mean global accuracy: 1.0
Final Evaluation Accuracy = 1.0

Is that the same with you?

**Comment by tensorfreitas** [comment]
Hi @basaltzhang . I am still working on the implementation. From my experience that occurs when the learning rate is too high, leading to the classifier predicting the same output for all images. 

With Adam optimizer I was able to get >80% accuracy, but with SGD+momentum as referred in the paper I am still trying to reproduce the results. 

When I end the implementation I will update it in the Readme. 

Despite that feel free to reach out if you have any doubts or if you find any bug!

**Comment by tensorfreitas** [comment]
I added some early stop conditions to avoid this issue, so i'll close this issue. 

Fell free to open it if you find it necessary! 

If you find another bug, do not hesitate to open another issue!
